webapp/tests/pluginsSelector-test.js

```diff
@@ -50,46 +55,67 @@ describe('plugin selectors', () => {
       }
     };
 
-    actual = sortPluginMetadata({ ...parentItems, ...subItems });
+    sortedPlugins = sortPluginMetadata({ ...parentItems, ...subItems });
   });
 
   describe('sortPluginMetadata', () => {
-    it('should return an array of only parent plugin metadata objects', () => {
-      const numberOfParents = Object.keys(parentItems).length;
-      expect(actual).to.have.lengthOf(numberOfParents);
-
-      let hasParents = false;
-      for (let pluginMeta of actual) {
-        hasParents = parentItems[pluginMeta.name] ? true : false;
-        expect(hasParents, `"${pluginMeta.name}" not in parents object`).to.be
-          .true;
-      }
+    it('should return an array of all plugins, parents and children', () => {
+      const totalPlugins =
+        Object.keys(parentItems).length + Object.keys(subItems).length;
+      expect(sortedPlugins).to.have.length(totalPlugins);
     });
 
     it('should call comparePlugins for sorting on parents and subItems', () => {
       const spy = sinon.spy(Array.prototype, 'sort');
       sortPluginMetadata({ ...parentItems, ...subItems });
       expect(spy.calledWith(plugins.comparePlugins)).to.be.true;
       const parentNames = Object.keys(parentItems).sort();
-      expect(parentNames[0]).to.equal(actual[0].name);
+      expect(parentNames[0]).to.equal(sortedPlugins[0].name);
     });
 
-    it('should add subItems to subItem property of parents', () => {
-      const subItemCount = {};
-      // get number of expected subItems for each parent
-      for (let plugin in subItems) {
-        const parent = subItems[plugin].parent;
-        let count = subItemCount[parent];
-        subItemCount[parent] = count === undefined ? 1 : count + 1;
-      }
+    it('should only have two level hierarchy, no children w/ children', () => {
+      const parents = new Set();
+      sortedPlugins.forEach(plugin => {
+        if (plugin.parent) parents.add(plugin.parent);
+      });
+      sortedPlugins.forEach(plugin => {
+        if (plugin.parent)
+          assert(
+            !parents.has(plugin.name),
+            `${plugin.name} has a parent and should not have children`
+          );
+      });
+    });
 
-      // confirm that each parent has expected number of subItems
-      for (let plugin of actual) {
-        if (subItemCount[plugin.name]) {
-          expect(plugin.subItems).to.not.be.undefined;
-          expect(plugin.subItems).to.have.lengthOf(subItemCount[plugin.name]);
+    it('should add sorted subItems after parent', () => {
+      const subItems = new Map();
```

Comment:
Can you add a comment around this control flow?

Reply:
heh, yeah this one was weird. Actually not thrilled that the first step basically does the same thing as what the actual function does to find the children in the first place, but it was tricky. Will add comments